pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -236,17 +263,75 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 		r.log.Error(err, "Managed object update failed, deleting and re-creating",
 			logFieldObject, dRef.String(),
 			logFieldKind, kind)
-		if err := r.client.Delete(ctx, declared); err != nil {
+		if err := deploymentResourceInterface.Delete(ctx, dRef.Name, metav1.DeleteOptions{}); err != nil {
 			return controllerutil.OperationResultNone, err
 		}
-		if err := r.client.Create(ctx, declared); err != nil {
+		data, err := json.Marshal(declared)
+		if err != nil {
+			return controllerutil.OperationResultNone, fmt.Errorf("failed to marshal declared deployment object to byte array: %w", err)
+		}
+		_, err = deploymentResourceInterface.Patch(ctx, dRef.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: reconcilermanager.ManagerName, Force: &forcePatch})
+		if err != nil {
 			return controllerutil.OperationResultNone, err
 		}
 	}
-
+	if deploymentAfterPatch.GetGeneration() == unObjCurrent.GetGeneration() {
```

Comment:
Do we also need to be checking the UID? If the UID changed then the generation MIGHT change, but it also might not, if it was 1 or something low.